### ISSUE TITLE
1.8.7 -> 1.9.0  - "successful" upgrade that does not start.

### ISSUE THREAD

**Issue by AndKe** [OctoPrint/OctoPrint#4810]
### The problem

Upgraded 1.8.7 -> 1.9.0rc6  
and this is where it ends:

![Screenshot from 2023-05-23 15-51-50](https://github.com/OctoPrint/OctoPrint/assets/1468842/73499dca-93aa-4ca2-a8de-b08185abcbca)

I had the same issue with earlier "release candidate"  - and solved it with
`$ ~/oprint/bin/pip install --force-reinstall OctoPrint==1.8.7 --no-cache-dir`

for details - see: https://community.octoprint.org/t/1-8-6-1-9-0rc2-went-very-bad-unable-to-start-octoprint-no-new-log-update-on-start-attempt/50435

### Did the issue persist even in safe mode?

Yes, it did persist

### Version of OctoPrint

1.9.0rc6 

### Operating system running OctoPrint

OctoPi

### Checklist of files to include below

- [X] Systeminfo Bundle (always include!)
- [ ] Contents of the JavaScript browser console (always include in cases of issues with the user interface)
- [ ] Screenshots and/or videos showing the problem (always include in case of issues with the user interface)
- [ ] GCODE file with which to reproduce (always include in case of issues with GCODE analysis or printing behaviour)

### Additional information & file uploads

more details and the systeminfo bundle (that did not help) at:
https://community.octoprint.org/t/1-8-6-1-9-0rc2-went-very-bad-unable-to-start-octoprint-no-new-log-update-on-start-attempt/50435

I'll wait a short while with reinstalling this time.

**Comment by AndKe** [comment]
dear nagbot - there is no "system info" bundle as the program does not start.

```$ ps -ef | grep -i octoprint | grep -i python
$
```

:)

**Comment by AndKe** [comment]
..and the same happens with the 1.9.0 release

**Comment by foosel** [comment]
Something seems to be seriously wrong with your install. I've gone through a ton of updates from various start versions throughout the whole RC phase, and I'm currently looking at almost 5k successfully upgraded and happily printing instances in the usage tracking.

The question is, *what* is wrong with your install that makes it completely throw away the whole OctoPrint files so that even the command cannot be found anymore.

At this point I'd honestly suggest to try if recreating the virtual environment helps, as a forced reinstall doesn't seem to do it. Alternative idea is that the SD card is dying or something like that, if that is running on one.

**Comment by AndKe** [comment]
@foosel  thanks.
I have upgraded, then returned to 1.8.7 many times now, (two times today, last with safe mode.)
I strongly doubt the high endurance SD to be that particulary corrupt.
I do love that every time I revert -. then things just works (all config/files are there)
Is there a way of doing a supercomplete backup? (with plugins and all?)  - then I could reflash the octopi too ? :)

**Comment by foosel** [comment]
Yes, the built in backup and restore plugin. Doesn't backup the plugin code, but stores what it can about the plugins you have installed and then attempts to reinstall all of them from the plugin repository. Plugin data is included in the backup.

**Comment by AndKe** [comment]
is it possible that my modified menu is to blame? 
![Screenshot from 2023-05-24 22-49-26](https://github.com/OctoPrint/OctoPrint/assets/1468842/c155bd9b-5292-473c-a296-c330324c054c)

**Comment by cp2004** [comment]
That should not be related. It's a feature of OctoPrint itself, not some unsuspecting modification (at least I hope you've done it properly).

**Comment by AndKe** [comment]
FWIW: exactly the same happened at work. (which has about the same plugins, but no menu customization.)
- solved by forcing the previous version.
These are the installed plugins:
![image](https://github.com/OctoPrint/OctoPrint/assets/1468842/b44f72bc-e005-4b0d-a26f-fd090f312060)

3rd party:
![image](https://github.com/OctoPrint/OctoPrint/assets/1468842/1c8a76ce-4212-479a-8db0-2e276e1660af)

**Comment by AndKe** [comment]
and systeminfo before upgrade:
[octoprint-systeminfo-20230525081621.zip](https://github.com/OctoPrint/OctoPrint/files/11562107/octoprint-systeminfo-20230525081621.zip)

**Comment by AndKe** [comment]
One system, the one at home, got clean installed using OctoPi image.
I can reproduce one last time the issue with the system at work - please specify exactly which logs you mean by "need to retrieve those logs after upgrade"

**Comment by jneilliii** [comment]
I'd first try to see if you can get to the recovery page by going to http://<ip>/recovery and see if it loads. If that doesn't then connecting via SSH to the pi and running the following commands to generate a bundle and then transfer that bundle from the pi with something like winscp/filezilla (if from windows machine). 

```
source ~/oprint/bin/activate
octoprint systeminfo .
```

More details about the process can be found [here](https://community.octoprint.org/t/what-is-a-systeminfo-bundle-and-how-can-i-obtain-one/29887?u=jneilliii).

**Comment by AndKe** [comment]
The job Octopi was the remaining one with this issue.  (which was reverted to 1.8.7)
I started with trying full backup.  - got an error that required to look into log to find the reason , which was octoprint.plugins.backup.InsufficientSpace

I did 'rm /home/pi/.octoprint/timelapse/*'
- made a backup , (516MB)
Then started the upgrade to 1.9.0   - and - this time ... it was completed successfully!.
I doubt the space was the issue, but have no idea what else it could be.   

Congrats on the recovery page! - very nice tool! - ..and quite hidden.  
Maybe it should be mentioned right in the "server not running" page and other, similar places.
